### PR TITLE
Add io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi

### PR DESCRIPTION
**Pull request by quyykk**
### Please confirm your submission meets all the criteria

This is an extension for [Endless Sky](https://flathub.org/apps/io.github.endless_sky.endless_sky), for a plugin containing high resolution images for the game (for people with 4k screens usually).

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. 
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. 
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. 

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by quyykk** [comment]
bot, build io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi

**Comment by quyykk** [comment]
bot, build io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi

File: io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  </description>
+  <url type="bugtracker">https://github.com/endless-sky/endless-sky-high-dpi/issues</url>
+  <url type="homepage">https://github.com/endless-sky/endless-sky-high-dpi/</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://private-user-images.githubusercontent.com/20605679/246690308-87ae00d1-43a8-4cf5-82a7-c50719dcacc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxMTM4NjYsIm5iZiI6MTcxMDExMzU2NiwicGF0aCI6Ii8yMDYwNTY3OS8yNDY2OTAzMDgtODdhZTAwZDEtNDNhOC00Y2Y1LTgyYTctYzUwNzE5ZGNhY2M1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEwVDIzMzI0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkYWRmMjMyMWFkODRjY2UzMTYyYzE3ZWM4NjM2YjE2NTI3ODM2NzM4NGI1MjcwYWI3OGM0YmQ5YjZiMmE1MzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lPA-v6Gpt5mUCUR4YhhqrWLnCQ_D7ooKySmEcvCBlTk</image>
```
Comment (bbhtt) [inline io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi.appdata.xml:17]: This screenshot url doesn't work, extensions don't need one, so checks are skipped. If this was an actual app, it'd fail validation.

**Comment by bbhtt** [comment]
> for a plugin containing high resolution images for the game

So this is just high-res assets for the game?

**Comment by quyykk** [comment]
Yes. On 4k screens the original assets are a bit blurry, so the high resolution assets are necessary. It's literally just every image but in double resolution.

File: io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi.appdata.xml (hunk 1 above)
Comment (quyykk) [reply io.github.endless_sky.endless_sky.plugins.endless_sky_high_dpi.appdata.xml]: Oh it seems to be a private URL for just my account. I didn't realize.

**Comment by hfiguiere** [comment]
why making this an extension?

Extension are a PITA and this doesn't justify it.

**Comment by quyykk** [comment]
Because it increases the size of the game flatpak by almost 4x, along with higher memory requirements.

**Comment by hfiguiere** [comment]
> , along with higher memory requirements. 

why would it if it doesn't use them?

**Comment by quyykk** [comment]
You're right it is not used by default. However, we've had cases in the past of people enabling it without having the required memory and as a result the game would just crash every time it's opened. The player would have to go look into the flatpak data and manually disable it, which is not a good experience in contrast to simply uninstalling the plugin through the store.

**Comment by quyykk** [comment]
> Extension are a PITA and this doesn't justify it.

May I ask why and why not?

**Comment by hfiguiere** [comment]
> Because it increases the size of the game flatpak by almost 4x, along with higher memory requirements.

and that's why it is auto installed.

ok.

**Pull request closed without merging**